tests/phpunit/CRM/Activity/Form/ActivityTest.php

```diff
@@ -240,34 +253,28 @@ private function deleteActivity($activityId, $mode = NULL) {
   }
 
   /**
-   * This is a bit messed up having a variable called name that means label but we don't want to fix it because it's a form member variable _activityTypeName that might be used in form hooks, so just make sure it doesn't flip between name and label. dev/core#1116
+   * This is a bit messed up having a variable called name that means label but
+   * we don't want to fix it because it's a form member variable
+   * _activityTypeName that might be used in form hooks, so just make sure it
+   * doesn't flip between name and label. dev/core#1116
+   *
+   * @throws \CRM_Core_Exception
    */
   public function testActivityTypeNameIsReallyLabel(): void {
-    $form = new CRM_Activity_Form_Activity();
-
-    // the actual value is irrelevant we just need something for the tested function to act on
-    $form->_currentlyViewedContactId = $this->source;
-
     // Let's make a new activity type that has a different name from its label just to be sure.
-    $actParams = [
-      'option_group_id' => 'activity_type',
+    $this->createTestEntity('OptionValue', [
+      'option_group_id:name' => 'activity_type',
       'name' => 'wp1234',
       'label' => 'Water Plants',
       'is_active' => 1,
+      'value' => 800,
       'is_default' => 0,
-    ];
-    $result = $this->callAPISuccess('option_value', 'create', $actParams);
-
-    $form->_activityTypeId = $result['values'][$result['id']]['value'];
-    $this->assertNotEmpty($form->_activityTypeId);
-
-    // Do the thing we want to test
-    $form->assignActivityType();
-
-    $this->assertEquals('Water Plants', $form->_activityTypeName);
+    ]);
 
-    // cleanup
-    $this->callAPISuccess('option_value', 'delete', ['id' => $result['id']]);
+    $form = $this->getTestForm('CRM_Activity_Form_Activity', [], ['atype' => 800, 'cid' => $this->source]);
```

Comment:
I get a fail here because action is null, so I think it just needs to add `'action' => 'add'` to the params. But also while it's nice addition to check the template vars, what this test was about was checking that somebody doesn't mess up `$form->_activityTypeName`.

Reply:
@demeritcowboy I've pushe dup the fix - but I am trying to avoid checking properties & check what the form outputs instead - the property is 'public' but ... not really. It's an internal form property - but what the form outputs is the variable....

Reply:
Since removing the existing coverage isn't necessary for fixing the bug and the discussion around public vars gets into scope creep, I'd suggest a compromise to add your test as a new test, which adds some good coverage, and then this would be good to merge (with the other test fail about the `action` url param addressed).

Reply:
@demeritcowboy actually the way I make it work to test the form is by switching to the form test wrapper - which gives a better level of testing - but it explicitly doesn't allow testing properties - other than where they have publicly supported getter methods